### DISCUSSION TITLE
Include YouTube video id in file name

### DISCUSSION THREAD

**Discussion by AlexGustafsson** [jmbannon/ytdl-sub#395] [Q&A] [answered]
When specifying a YouTube channel as a subscription, how does one ensure that the video's id is mentioned in the filename?

I'm trying to use the recommended format to work with https://github.com/ankenyr/jellyfin-youtube-metadata-plugin.

An example:

```
3Blue1Brown - 20211023 - A_few_of_the_best_math_explainers_from_this_summer [F3Qixy-r_rQ].mkv
```

My current attempt is

```
file_name: "{channel} - {upload_year}{upload_month_padded}{upload_day_padded} - {title_sanitized} [{source_uid}]"
```

Where `channel` is specified for each subscription.

The value for `source_uid` doesn't seem to be available, because it is replaced with the default - the "playlist's" (channel's) uid instead.

**Comment by jmbannon** [comment, marked as answer]
Use `{uid}`: https://ytdl-sub.readthedocs.io/en/latest/config.html#ytdl_sub.entries.variables.entry_variables.EntryVariables.uid

**Reply by AlexGustafsson** [reply]
Works like a charm. Thanks!
